pkg/build/interfaces.go

```diff
@@ -0,0 +1,50 @@
+package build
+
+import "github.com/openshift/source-to-image/pkg/api"
+
+// Builder is the interface that provides basic methods all implementation
+// should have.
+// Build method executes the build based on Request and returns the Result.
+type Builder interface {
+	Build(*api.Request) (*api.Result, error)
+}
+
+// Preparer provides the Prepare method for builders that need to prepare source
+// code before it gets passed to the build. All builders must comfort this
```

Comment:
also if builders must implement it, why not make it part of the Builder interface?